crates/sql-sink/src/main.rs

```diff
@@ -7,69 +7,100 @@ mod upsert;
 
 use std::time::Duration;
 
+use adaptive_backoff::prelude::{
+    Backoff, BackoffBuilder, ExponentialBackoff, ExponentialBackoffBuilder,
+};
 use anyhow::{anyhow, Result};
 use config::SqlConfig;
 use futures::{SinkExt, StreamExt};
 
 use fluvio_connector_common::{
     connector,
     consumer::ConsumerStream,
-    future::retry::ExponentialBackoff,
-    tracing::{error, trace, warn},
+    tracing::{error, info, trace, warn},
     LocalBoxSink, Sink,
 };
 use fluvio_model_sql::Operation;
 
 use sink::SqlSink;
 
 const BACKOFF_MIN: Duration = Duration::from_secs(1);
-const BACKOFF_MAX: Duration = Duration::from_secs(3600 * 24);
+const BACKOFF_MAX: Duration = Duration::from_secs(600);
 
 #[connector(sink)]
 async fn start(config: SqlConfig, mut stream: impl ConsumerStream) -> Result<()> {
-    let mut sink = start_sink(config).await?;
-    while let Some(item) = stream.next().await {
-        let operation: Operation = serde_json::from_slice(item?.as_ref())?;
-        trace!(?operation);
-        sink.send(operation).await?;
+    let mut backoff = backoff_init()?;
+    let mut sink = start_sink(&mut backoff, &config).await?;
+
+    info!("Starting to process records");
+
+    while let Some(item_result) = stream.next().await {
+        match item_result {
+            Ok(item) => match serde_json::from_slice::<Operation>(item.as_ref()) {
+                Ok(operation) => {
+                    trace!(?operation);
+                    while let Err(e) = sink.send(operation.clone()).await {
+                        error!("Error trying to send to db: {}", e);
+                        sink = start_sink(&mut backoff, &config).await?;
+                    }
+                    backoff.reset();
+                }
+                Err(err) => {
+                    error!("Failed to deserialize operation: {}", err);
+                    backoff_and_wait(&mut backoff).await?;
+                }
+            },
+            Err(err) => {
+                error!("Error reading from stream: {}", err);
+                backoff_and_wait(&mut backoff).await?;
+            }
+        }
     }
 
+    info!("Connection dropped, shutting down");
+
     Ok(())
 }
 
-async fn start_sink(config: SqlConfig) -> Result<LocalBoxSink<Operation>> {
-    let mut backoff = backoff_init()?;
+async fn backoff_and_wait(backoff: &mut ExponentialBackoff) -> Result<()> {
+    let wait = backoff.wait();
+    if wait < BACKOFF_MAX {
+        warn!(
+            "Waiting {} before next attempting to db",
+            humantime::format_duration(wait)
+        );
+        async_std::task::sleep(wait).await;
+        Ok(())
+    } else {
+        let err_msg = "Max retry wait time exceeded, shutting down";
```

Comment:
change this to something like: 
```
max retry on SQL execution
```

This is related to one of the log in the issue: https://github.com/infinyon/fluvio/issues/4265

Reply:
Sure, fixed. 

The bug that it was creating big log files? Yeah, that happened to me, it was more than 100gb of my logs. 
Now with sleep this is not happening.